macos/FluentUI/Button/Button.swift

```diff
@@ -121,6 +122,37 @@ open class Button: NSButton {
 		}
 	}
 
+	func setupBorderShadowsIfNeeded() {
+		// At the moment border shadows only apply for the Primary and Secondary Button Styles.
+		// Border shadows being meant for buttons and their corresponding States, that have ~some~
+		// background color other than `clear` which is why they don't apply to the Borderless Style
+		// buttons. Acrylic Style buttons do possess a background color, yet they maintain transparency.
+		// This transparency allows any underlying border shadow layers to bleed through, consequently
+		// impacting the overall button background color. Adding shadows in this context presents a
+		// challenge. Therefore, unless it becomes necessary, these buttons will remain without shadows.
+		// And perhaps shadowPath is the way to go, where you can have the CALayer have a Clear
+		// background color and still show a shadow by setting a hollow shadowPath around the button.
+		self.usesBorderShadows = style == .primary || style == .secondary
```

Comment:
2 minor things:
1. can omit `self` for `self.usesBorderShadows`
2. Tend to prefer switch/case statements when dealing with enums. That forces new enum values to explicitly make behavioral decisions at compile time, which generally makes that experience much easier.